src/routes/CreateSafePage/components/SafeCreationProcess.tsx

```diff
@@ -78,102 +87,118 @@ const parseError = (err: Error): Error => {
-        const safeCreationSalt = safeCreationFormValues[FIELD_NEW_SAFE_PROXY_SALT]
-        const gasLimit = safeCreationFormValues[FIELD_NEW_SAFE_GAS_LIMIT]
-        const gasPrice = safeCreationFormValues[FIELD_NEW_SAFE_GAS_PRICE]
-        const gasMaxPrioFee = safeCreationFormValues[FIELD_NEW_SAFE_GAS_MAX_PRIO_FEE]
-        const deploymentTx = getSafeDeploymentTransaction(ownerAddresses, confirmations, safeCreationSalt)
-
-        const sendParams = createSendParams(userAddressAccount, {
-          ethGasLimit: gasLimit.toString(),
-          ethGasPriceInGWei: gasPrice,
-          ethMaxPrioFeeInGWei: gasMaxPrioFee.toString(),
+  if (!safeCreationFormValues) {
+    return Promise.reject(new Error('No saved Safe creation'))
+  }
+
+  return new Promise((resolve, reject) => {
+    const confirmations = safeCreationFormValues[FIELD_NEW_SAFE_THRESHOLD]
+    const ownerFields = safeCreationFormValues[FIELD_SAFE_OWNERS_LIST]
+    const ownerAddresses = ownerFields.map(({ addressFieldName }) => safeCreationFormValues[addressFieldName])
+    const safeCreationSalt = safeCreationFormValues[FIELD_NEW_SAFE_PROXY_SALT]
+    const gasLimit = safeCreationFormValues[FIELD_NEW_SAFE_GAS_LIMIT]
+    const gasPrice = safeCreationFormValues[FIELD_NEW_SAFE_GAS_PRICE]
+    const gasMaxPrioFee = safeCreationFormValues[FIELD_NEW_SAFE_GAS_MAX_PRIO_FEE]
+    const deploymentTx = getSafeDeploymentTransaction(ownerAddresses, confirmations, safeCreationSalt)
+
+    const sendParams = createSendParams(userAddress, {
+      ethGasLimit: gasLimit.toString(),
+      ethGasPriceInGWei: gasPrice,
+      ethMaxPrioFeeInGWei: gasMaxPrioFee.toString(),
+    })
+
+    deploymentTx
+      .send(sendParams)
+      .once('transactionHash', (txHash) => {
+        onHash(txHash)
+
+        saveToStorage(SAFE_PENDING_CREATION_STORAGE_KEY, {
+          [FIELD_NEW_SAFE_CREATION_TX_HASH]: txHash,
+          ...safeCreationFormValues,
         })
 
-        deploymentTx
-          .send(sendParams)
-          .once('transactionHash', (txHash) => {
-            saveToStorage(SAFE_PENDING_CREATION_STORAGE_KEY, {
-              [FIELD_NEW_SAFE_CREATION_TX_HASH]: txHash,
-              ...safeCreationFormValues,
-            })
-
-            // Monitor the latest block to find a potential speed-up tx
-            txMonitor({ sender: userAddressAccount, hash: txHash, data: deploymentTx.encodeABI() })
-              .then((txReceipt) => {
-                console.log('Speed up tx mined:', txReceipt)
-                resolve(txReceipt)
-              })
-              .catch((error) => {
-                reject(parseError(error))
-              })
-          })
+        // Monitor the latest block to find a potential speed-up tx
+        txMonitor({ sender: userAddress, hash: txHash, data: deploymentTx.encodeABI() })
           .then((txReceipt) => {
-            console.log('First tx mined:', txReceipt)
+            console.log('Speed-up tx mined:', txReceipt)
```

Comment:
```suggestion
            console.log('Speed-up mined tx:', txReceipt)
```

Reply:
It's not in imperative, it's a noun phrase. "Original tx [has been] mined" vs "Speed-up tx [has been] mined" (as in, a tx that tries to speed up the original tx"). I can change it to "sped-up" for it to look more like an adjective.